PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEDs.cxx

```diff
@@ -1013,36 +1013,52 @@ void AliAnalysisTaskSEDs::UserExec(Option_t * /*option*/)
 
         if (fApplyML)
         {
-          isMLsel = fMLResponse->IsSelected(modelPred, d, aod->GetMagneticField(), Pid_HF, 0);
+            isMLsel = fMLResponse->IsSelectedMultiClass(modelPred, d, aod->GetMagneticField(), Pid_HF);
+
+          if (isMLsel)
+          {
+            if (modelPred.size() == 0)
+            {
+              modelPred.push_back(-1.);
+              modelPred.push_back(-1.);
+              modelPred.push_back(-1.);
+            }
```

Comment:
Why do you should have this case? If `isMLsel` is true `modelPred` should be filled.

Reply:
`isMLsel` is true also if no ML is applied and in this case if ve don't fill `modelPred` with three dummy values we get an error when we fill `var4nSparse`, so I would say that it is ok

Reply:
Effectively, `isMLsel` is used only if we are applying the model. See https://github.com/alisw/AliPhysics/blob/05b88d37eb04282ac4af04305501f7e9d8096ff5/PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEDs.cxx#L1048
probably we could also initialize it to false.
Since now `var4nSparse` is a std::vector we could just insert the content of `modelPred` (without the need to add dummy values) in it. Then, the length of the vector is different depending on where we are using ML or not and multiclass or not but it will match the THnSparse structure by default.
So, we could remove L1018-1031 and initialize the vector like this:
```
std::vector<Double_t> var4nSparsePID = {..., ..., ...};
var4nSparsePID.insert(var4nSparsePID.begin(), modelPred.begin(), modelPred.end())
```

Reply:
Indeed this solution would also work, but you mean `var4nSparse` and not `var4nSparsePID`, since the latter is only filled in case of `fApplyML` (and it's not a `std::vector`), is it correct?

Reply:
In this PR also `var4nSparsePID` is a `std::vector` https://github.com/alisw/AliPhysics/blob/820a05353136d1c245223975b6ce59e3d1712f4a/PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEDs.cxx#L1035

Reply:
Right, so to recap for @stefanopolitano:
- initialise `isMLsel` to `false`
- remove the all the `push_back(-1.)` both in case of null size and size equal unity
- initialise the `var4nSparse`, `var4nSparsePID`, and `var4nSparsePIDcomb` as @fcatalan92 suggested

Reply:
I agree with Fabio, this could be a good way to fill var4nSparse according to the size of modelPred without introducing dummy values and we could use it when we have to insert the ML output in a vector (var4nSparse, var4nSparsePIDcomb). 
However, instead of:
   var4nSparsePID.insert(var4nSparsePID.begin(), modelPred.begin(), modelPred.end())
I would use:
   var4nSparsePID.insert(var4nSparsePID.end(), modelPred.begin(), modelPred.end())
In this way, we could have ML outputs at the end of the vector(s): if it is binary classification, we can simply reduce by 2 the number of axes as Fabrizio was suggesting. What do you think?

Reply:
Yes I agree too! I would also do as you suggest in order to have the ML outputs at the end of the final vector and reduce the number of axes in case of binary classification. Thanks!

Reply:
Yes, indeed it was a typo! You should add them at the end in order to to use the trick.